ConvertTsv.py

```diff
@@ -2,96 +2,119 @@
+
+            keys.append(os.path.splitext(basename)[0])
+            texts.append(''.join(f.readlines()[2:]))
+            labels.append(os.path.split(filepath)[-1])
+
+    news = pd.DataFrame({'key': keys, 'label': labels, 'text': texts})
+
+    # natoo-pyではコンストラクタに渡す
+
+    mecab_sym = MeCab("-Ochasen")
+    print(mecab_sym.parse('こんにちはかなちゃん。'))
+    a = 'こんにちはかなちゃん。'
+    print(mecab_sym.parse(a).split("\n"))
+    symbol_list = []
+
+    def pick_sym(text):
+        node = mecab_sym.parse(text)
+        node_list = node.split("\n")
+        for node in node_list:
+            if '記号' in node:
+                symbol_list.append(node[0])
+
+    news['text'] = news['text'].apply(mojimoji.han_to_zen)
+
+    symbol_list = []
+    for i in range(len(news)):
+        symbol_list.append(pick_sym(news['text'].iloc[i]))
+
+    symbol_list = list(set(symbol_list))
+    print(symbol_list)
+
+    def del_sym(df):
+        # print(df['text'])
+        df['text'] = df['text'].replace('０', '')
+        df['text'] = df['text'].replace('１', '')
+        df['text'] = df['text'].replace('２', '')
+        df['text'] = df['text'].replace('３', '')
+        df['text'] = df['text'].replace('４', '')
+        df['text'] = df['text'].replace('５', '')
+        df['text'] = df['text'].replace('６', '')
+        df['text'] = df['text'].replace('７', '')
+        df['text'] = df['text'].replace('８', '')
+        df['text'] = df['text'].replace('９', '')
+        df['text'] = df['text'].replace('\d+年', '')
+        df['text'] = df['text'].replace('\d+月', '')
+        df['text'] = df['text'].replace('\d+日', '')
+        df['text'] = df['text'].replace('\d', '')
+        df['text'] = df['text'].replace('\n', '')
+
+        for i in range(len(symbol_list)):
+            if not symbol_list[i]:
+                continue
+            df['text'] = df['text'].replace(symbol_list[i], '')
+        return df
 
-    f.close()
+    # いらないものを取り去り、tsvに格納する作業
+    for DataFrameTextIndex in range(len(news)):
+       #  print(news.iloc[DataFrameTextIndex])
+        news.iloc[DataFrameTextIndex] = del_sym(news.iloc[DataFrameTextIndex])
 
-    # ここから前処理
-    def preprocessing_text(text):
+    news['label'] = news['label'].replace({'it-life-hack': 0, 'kaden-channel': 1, })
 
-        text = re.sub('<br />', '', text)
+    # train_set, test_set = train_test_split(news, test_size=0.2)
+    train_set, test_set = train_test_split(news, test_size=0.2)
 
-        for p in string.punctuation:
-            if (p == ".") or (p == ","):
-                continue
-        text = text.replace(p, " ")
-
-        text = text.replace(".", " . ")
-        text = text.replace(",", " , ")
-        return text
+    f = open(base + '/data-japanese/jp_train.tsv', 'w')
+    for index in range(len(train_set)):
+        text = train_set['text'].iloc[index]
+        label_num = train_set['label'].iloc[index]
+        text = text + '\t' + str(label_num) + '\t' + '\n'
+        f.write(text)
+    f.close()
```

Comment:
`ファイルを開く->何らかの処理->処理内容を書き込む->ファイルを閉じる` は、この程度の処理ですとcloseし忘れないですが、人間はcloseし忘れる人間なので、 `何らかの処理->ファイルを開く->書き込む->ファイルを閉じる` という流れをおすすめします。このとき、 `with` 構文を利用するとブロックを抜けたときに自動でファイルが閉じられるのでおすすめです。

また先述の通り、pandasのDataFrameをfor文で扱うのは速度の観点からおすすめできません（この程度なら無視できるかもしれませんが、今後よりサイズの大きいデータセットを扱うことを想定して、コメントします。）

Reply:
そして、ここの処理ですが、結論から言うと `pd.to_csv` を使うと一行で書くことが可能です。`news` の中の `text` カラムと`label` カラムをTSVで出力する例です:
```suggestion
    news[['text', 'label']].to_csv(base + '/data-japanese/jp_train.tsv', delimiter='\t')
```

Reply:
「delimiter」は使用できませんでしたが、sep='\t'を指定することで同じ操作を行うことができました

Reply:
```sep='\t' , index=False```
の指定でindexを取り除くことができました。
自分の場合はindexは必要にならないので消しました。